src/ansiblelint/rules/risky_octal.md

```diff
@@ -1,6 +1,11 @@
 # risky-octal
 
-This rule checks that octal file permissions either contain a leading zero or are a string value.
+This rule checks that octal file permissions are strings that contain a leading
+zero or are written in symbolic mode (such as `u+rwx` or `u=rw,g=r,o=r`).
+
+This is because using a numeric decimal or octal in YAML can easily lead to
+costly unexpected mistakes. For example, `0644` is interpreted as `420` as
+a decimal.
```

Comment:
I'd like to clarify this bit some more. Is this intended to recommend symbolic mode instead of octal number?

When would the YAML parser interpret 0644 as the 420 decimal number? Does that happen during serialization or something?

Reply:
It is happening during the loading, so is outside our control.

Reply:
So is it recommended to use symbolic mode instead of something like 0644 to avoid the wrong interpretation?

Reply:
From our point of view using a string with `0644` is equally safe as symbolic mode. User can decide. In the future we might even suggest how symbolic representation for the octal, so they can switch from one to another. Still, for now we only want to avoid integers/octals inside the YAML, as they are tricky.